ClientTests/FeatureFlagManagerTests.swift

```diff
@@ -0,0 +1,228 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+import XCTest
+
+@testable import Client
+
+class FeatureFlagManagerTests: XCTestCase, FeatureFlaggable {
+
+    // MARK: - Test Lifecycle
+    override func setUp() {
+        super.setUp()
+        let mockProfile = MockProfile(databasePrefix: "FeatureFlagsManagerTests_")
+        mockProfile.prefs.clearAll()
+        FeatureFlagsManager.shared.initializeDeveloperFeatures(with: mockProfile)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+    func testExpectedCoreFeatures() {
+        let adjustSetting = featureFlags.isCoreFeatureEnabled(.adjustEnvironmentProd)
+        let mockDataSetting = featureFlags.isCoreFeatureEnabled(.useMockData)
+        let contileAPISetting = featureFlags.isCoreFeatureEnabled(.useStagingContileAPI)
+
+        #if MOZ_CHANNEL_RELEASE
+            XCTAssertTrue(adjustSetting)
+            XCTAssertFalse(mockDataSetting)
+            XCTAssertFalse(contileAPISetting)
+        #elseif MOZ_CHANNEL_BETA
+            XCTAssertTrue(adjustSetting)
+            XCTAssertFalse(mockDataSetting)
+            XCTAssertTrue(contileAPISetting)
+        #elseif MOZ_CHANNEL_FENNEC
+            XCTAssertFalse(adjustSetting)
+            XCTAssertTrue(mockDataSetting)
+            XCTAssertTrue(contileAPISetting)
+        #else
+            XCTFail("We are in an unknown build type. WHat's happening!? How did we get here?!")
+        #endif
+    }
+
+    func testDefaultNimbusBoolFlags() {
+        // Tests for default settings should be performed on both build and user
+        // prefs separately to ensure that we are getting the expected results on both.
+        // Technically, at this stage, these should be the same.
+        #if MOZ_CHANNEL_RELEASE
```

Comment:
How would that be changed for tests? We are always on the same channel when testing no?

Reply:
Technically always developer, yeah. But I didn't know what's happening on BR. Like if we're building for Beta, are we testing anything? WHat about release? If we don't, then.... yeah, this is probably irrelevant.

Reply:
We don't run unit tests when releasing Beta, Nightly or Release. At least I don't see it in the BR workflows!